### ISSUE TITLE
Native crash at /system/vendor/lib/egl/libGLESv2_adreno.so

### ISSUE THREAD

**Issue by westnordost** [tangrams/tangram-es#1460]
I have so many users now, StreetComplete is becoming a stress test for tangram-es ;-)

So, here is another crash report I get via the Android developer console. It happened on the following devices and Android versions so far:

Android 5.0, 5.1
Devices: OnePlus One, Galaxy Note3, Galaxy S4

One user added the following comment:
> "The app crashed after I turned off the smartphone with the power button and then turned it on again."

```
*** *** *** *** *** *** *** *** *** *** *** *** *** *** *** ***
Build fingerprint: 'samsung/jfltexx/jflte:5.0.1/LRX22C/I9505XXUHPK2:user/release-keys'
Revision: '11'
ABI: 'arm'
pid: 18988, tid: 19025, name: RenderThread >>> de.westnordost.streetcomplete <<<
signal 11 (SIGSEGV), code 1 (SEGV_MAPERR), fault addr 0xe
r0 00000000 r1 00004001 r2 00000000 r3 a0a181e8
r4 a0a181e8 r5 0000000e r6 00000078 r7 a0a41000
r8 00000000 r9 00000005 sl 9850c980 fp a0a404c0
ip ad5ddea8 sp a0cd3788 lr b6f65293 pc a950fa40 cpsr 600d0030

backtrace:
#00 pc 00080a40 /system/vendor/lib/egl/libGLESv2_adreno.so (cpumempool_malloc+67)
#01 pc 0008c1f1 /system/vendor/lib/egl/libGLESv2_adreno.so (rb_vbo_alloc+16)
#02 pc 00050e49 /system/vendor/lib/egl/libGLESv2_adreno.so (cache_temp_vbo_elements+12)
#03 pc 00054885 /system/vendor/lib/egl/libGLESv2_adreno.so (core_glDrawElementsInstancedXXX+440)
#04 pc 000549b1 /system/vendor/lib/egl/libGLESv2_adreno.so (core_glDrawElements+10)
#05 pc 00043ca3 /system/vendor/lib/egl/libGLESv2_adreno.so (glDrawElements+28)
#06 pc 000283d1 /system/lib/libhwui.so
#07 pc 000283fb /system/lib/libhwui.so
#08 pc 000285c1 /system/lib/libhwui.so
#09 pc 000322ff /system/lib/libhwui.so
#10 pc 0001caf9 /system/lib/libhwui.so
#11 pc 0001c6c3 /system/lib/libhwui.so
#12 pc 0002a2ab /system/lib/libhwui.so
#13 pc 0003dd39 /system/lib/libhwui.so
#14 pc 0003ec45 /system/lib/libhwui.so
#15 pc 00040743 /system/lib/libhwui.so (_ZN7android10uirenderer12renderthread12RenderThread10threadLoopEv+66)
#16 pc 0000ef5d /system/lib/libutils.so (_ZN7android6Thread11_threadLoopEPv+112)
#17 pc 0005f391 /system/lib/libandroid_runtime.so (_ZN7android14AndroidRuntime15javaThreadShellEPv+72)
#18 pc 0000eacd /system/lib/libutils.so
#19 pc 000137a3 /system/lib/libc.so (_ZL15__pthread_startPv+30)
#20 pc 00011883 /system/lib/libc.so (__start_thread+6)
```

**Comment by westnordost** [comment]
(added more info)

**Comment by hjanetzek** [comment]
From the backtrace one cannot infer a direct relation to tangram. It could be a bug in Android's  libGLESv2_adreno.so. I've seen all kind of weird workarounds to avoid to trigger such bugs.. (e.g. google 'android 5 libgles sigsegv')

Thanks for the info though - We'll see if we can reproduce the crash and find a fix.

**Comment by westnordost** [comment]
Yeah, I thought so. I figured I just forward this to you just in case it is helpful to you, since tangram is the only component in my app that uses gles (that I know of).

On 19 April 2017 10:47:52 CEST, Hannes Janetzek <notifications@github.com> wrote:
>>From the backtrace one cannot infer a direct relation to tangram. It
>could be a bug in Android's  libGLESv2_adreno.so. I've seen all kind of
>weird workarounds to avoid to trigger such bugs.. (e.g. google 'android
>5 libgles sigsegv')
>
>Thanks for the info though - We'll see if we can reproduce the crash
>and find a fix.
>
>
>
>-- 
>You are receiving this because you authored the thread.
>Reply to this email directly or view it on GitHub:
>https://github.com/tangrams/tangram-es/issues/1460#issuecomment-295169231

-- 
Diese Nachricht wurde von meinem Android-Mobiltelefon mit K-9 Mail gesendet.

**Comment by westnordost** [comment]
Since the last update, this crash was not reported anymore. I think it might have been fixed with #1469
